### DISCUSSION TITLE
Deepstream Recipe from meta-tegra-community reaches 1.5 GB

### DISCUSSION THREAD

**Discussion by Islam-Hussein-11** [OE4T/meta-tegra#1471] [Q&A]
Hi,
I'm trying to keep my Image as small as possible. I found out that deepstream-6.3 takes 1.5 GB which is too costy as in further development I'll need updates and second Partition so I calculate everything by factor of 200%. So If it is like this it will be 3 GB for deepstream recipe only. Is there a way to reduce this size as much as possible. And I made sure that this is the bare recipe without -samples/-samples-data/-dev/-dbg or anything. 

In packages-split folder I found out that deepstream-6.3 folder took about 57 MB and samples  or samples-data were not shipped with the Image. 

So if anyone could enlighten me and tell me the cause of that problem ??
![deepstream-6 3-Size](https://github.com/OE4T/meta-tegra/assets/153657677/a86458d2-9f7f-4390-bf03-2117158f2b1f)

**Comment by madisongh** [comment]
When you say that "deepstream-6.3 takes 1.5 GB", you're talking about the size of the full root filesystem, correct?  The DS libraries themselves may not be very large, but they have dependencies on things like CuDNN and TensorRT, and cuBLAS that do have some very large libraries.  Even if you drop DS itself, you're likely going to need those other library packages to implement any sort of machine learning application that uses CUDA. You should definitely plan on using a multi-gigabyte rootfs partition size for any such project.

**Reply by Islam-Hussein-11** [reply]
I'd like to thank you for your support so far. I just made an .bbappend that includes another package that includes the .so files that we need exactly and packed the rest to the main package. It didn't make so much difference but I've squeezed it at most.
